spark/v3.2/spark/src/test/java/org/apache/iceberg/spark/source/TestSparkCatalogCacheExpiration.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.spark.source;
+
+import java.util.Map;
+import org.apache.iceberg.CatalogProperties;
+import org.apache.iceberg.relocated.com.google.common.collect.ImmutableMap;
+import org.apache.iceberg.spark.SparkCatalog;
+import org.apache.iceberg.spark.SparkCatalogTestBase;
+import org.apache.spark.sql.connector.catalog.TableCatalog;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runners.Parameterized;
+
+public class TestSparkCatalogCacheExpiration extends SparkCatalogTestBase {
+
+  private final String catalogConfigPrefix;
+  private final boolean isCacheEnabled;
+  private final boolean isCacheExpirationEnabled;
+
+  @Parameterized.Parameters(name = "catalogName = {0}, implementation = {1}, config = {2}")
+  public static Object[][] parameters() {
+    return new Object[][] {
+        { "testcacheexpirationnotenabled", SparkCatalog.class.getName(),
+          ImmutableMap.of(
+            "type", "hive",
+            "default-namespace", "default",
+            CatalogProperties.TABLE_CACHE_ENABLED, "false",
+            CatalogProperties.TABLE_CACHE_EXPIRATION_ENABLED, "false"
+        ) },
+        { "testhadoop", SparkCatalog.class.getName(),
+           ImmutableMap.of(
+            "type", "hadoop",
+            CatalogProperties.TABLE_CACHE_ENABLED, "true",
+            CatalogProperties.TABLE_CACHE_EXPIRATION_ENABLED, "false"
+         ) }
+    };
+  }
+
+  public TestSparkCatalogCacheExpiration(String catalogName,
+                                         String implementation,
+                                         Map<String, String> config) {
+    super(catalogName, implementation, config);
+    this.catalogConfigPrefix = "spark.sql.catalog." + catalogName + ".";
+    this.isCacheEnabled = Boolean.parseBoolean(
+        config.getOrDefault(CatalogProperties.TABLE_CACHE_ENABLED, "true"));
+    this.isCacheExpirationEnabled = Boolean.parseBoolean(
+        config.getOrDefault(CatalogProperties.TABLE_CACHE_EXPIRATION_ENABLED, "false"));
+  }
+
+  // TODO - Looking for a better way to test this. Not 100% sure how to access the Iceberg catalog
```

Comment:
you could try something like `Assertions.assertThat(catalog).extracting("icebergCatalog").isInstanceOf(CachingCatalog.class);`

Reply:
so basically your checks could become something like this:
```
Assertions.assertThat(catalog).extracting("icebergCatalog").isInstanceOf(CachingCatalog.class);
Assertions.assertThat(catalog).extracting("icebergCatalog").extracting("expirationIntervalMillis").isEqualTo(0);
Assertions.assertThat(catalog).extracting("icebergCatalog").extracting("expirationEnabled").isEqualTo(true);
Assertions.assertThat(catalog).extracting("cacheEnabled").isEqualTo(true)
```

Reply:
Oooo that's great. I was not aware of this `extracting` business. Thanks for sharing, Eduard!